### DISCUSSION TITLE
X axis labels not starting at 0.0 like y axis

### DISCUSSION THREAD

**Discussion by MathijsWouters** [Live-Charts/LiveCharts2#1585] [Q&A]
So i am writing a program that is using this library in maui i just changed from a line graph to a bar graph and for the life of me i can't figure out why the x axes labels do not align with their bar. 

```
using LiveChartsCore;
using LiveChartsCore.SkiaSharpView;
using LiveChartsCore.SkiaSharpView.Painting;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;

namespace BeursfuifChiroZevendonk.ViewModels
{
    public partial class BeursPageViewModel : BaseViewModel
    {
        public ObservableCollection<Drink> Drinks => _drinksService.Drinks;
        public ObservableCollection<ISeries> Series { get; set; } = new ObservableCollection<ISeries>();

        private readonly DrinksDataService _drinksService;
        public IEnumerable<Axis> XAxes { get; set; }
        public IEnumerable<Axis> YAxes { get; set; }

        public BeursPageViewModel(DrinksDataService drinksService)
        {
            base.Title = "Beurs";
            _drinksService = drinksService;
            MessagingCenter.Subscribe<App>((App)Application.Current, "PricesUpdated", (sender) => {
                UpdateChart();
            });
            UpdateChart();
        }

        private void InitializeAxes()
        {
            decimal highestMaxPrice = Drinks.Max(drink => drink.MaxPrice) + 0.25m;
            decimal lowestMinPrice = Drinks.Min(drink => drink.MinPrice) - 0.25m;

            YAxes = new Axis[]
            {
        new Axis
        {
            MinLimit = (double)lowestMinPrice,
            MaxLimit = (double)highestMaxPrice,
            ShowSeparatorLines = false,
            LabelsPaint = new SolidColorPaint(SKColors.White),
            TextSize = 20
        }
            };

            XAxes = new Axis[]
                {
                    new Axis
                    {
                        Labels = Drinks.Select(d => d.Name).ToArray(),
                        LabelsPaint = new SolidColorPaint(SKColors.White),
                        TextSize = 20,
                        LabelsRotation = 0,
                        SeparatorsPaint = new SolidColorPaint(SKColors.Transparent)
                    }
                };
        }


        public void UpdateChart()
        {
            InitializeAxes();
            var series = new List<ISeries>();

            foreach (var drink in Drinks)
            {
                var columnSeries = new ColumnSeries<decimal>
                {
                    Values = new decimal[] { drink.CurrentPrice },
                    Name = drink.Name,
                    Fill = new SolidColorPaint(SKColor.Parse(drink.DrinkColorHex)),
                    MaxBarWidth = double.NaN, 
                    Padding = 50, 
                };

                series.Add(columnSeries);
            }

            Series.Clear();
            foreach (var serie in series)
            {
                Series.Add(serie);
            }

            RefreshChart();
        }

        private void RefreshChart()
        {
            OnPropertyChanged(nameof(Series));
            OnPropertyChanged(nameof(XAxes));
            OnPropertyChanged(nameof(YAxes));
        }
    }
}
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:vm="clr-namespace:BeursfuifChiroZevendonk.ViewModels"
             xmlns:models="clr-namespace:BeursfuifChiroZevendonk.Models"
             xmlns:lvc="clr-namespace:LiveChartsCore.SkiaSharpView.Maui;assembly=LiveChartsCore.SkiaSharpView.Maui"
             x:Class="BeursfuifChiroZevendonk.Views.BeursPage"
             x:DataType="vm:BeursPageViewModel"
             Title="{Binding Title}">

    <Grid RowDefinitions="3*,*">
        <lvc:CartesianChart Series="{Binding Series}"
                    YAxes="{Binding YAxes}"
                    XAxes="{Binding XAxes}"
                            Margin="0"
                            Padding="0"
                            >
        </lvc:CartesianChart>
        <ScrollView Grid.Row="1" Orientation="Horizontal">
            <CollectionView ItemsSource="{Binding Drinks}" 
                            HorizontalOptions="CenterAndExpand">
                <CollectionView.ItemsLayout>
                    <LinearItemsLayout Orientation="Horizontal" ItemSpacing="10" />
                </CollectionView.ItemsLayout>
                <CollectionView.ItemTemplate>
                    <DataTemplate x:DataType="models:Drink">
                        <Frame BorderColor="White" CornerRadius="10" Padding="0" Margin="5,2.5" WidthRequest="150">
                            <Grid RowDefinitions="*,*,*">
                                <Label Grid.Row="0"
                           Text="{Binding Name}"
                           TextColor="White"
                           FontAttributes="Bold"
                           HorizontalOptions="Center"
                           VerticalOptions="Center"
                                       FontSize="Medium"/>
                                <Label Grid.Row="1"
                           Text="{Binding CurrentPrice, StringFormat='€{0:0.00}'}"
                           TextColor="White"
                           FontAttributes="Bold"
                           HorizontalOptions="Center"
                           VerticalOptions="Center"
                                       FontSize="Medium"/>
                                <BoxView Grid.Row="2"
                             Color="{Binding DrinkColor}"
                             Opacity="1"/>
                            </Grid>
                        </Frame>
                    </DataTemplate>
                </CollectionView.ItemTemplate>
            </CollectionView>
        </ScrollView>
    </Grid>
</ContentPage>

```
![Schermafbeelding 2024-09-02 143154](https://github.com/user-attachments/assets/c14587e5-19d7-412d-aaff-a6da9a8b9677)

Please someone help

**Comment by beto-rodriguez** [comment]
Hi!

That is too much code!

Is there a minimal reproducible sample where I can take a look at?

I don't see any label in your chart, what am i missing?

I will close this for now but will move it to a discussion where you can provide more info about the issue you are facing and luckily someone will be able to help you, if you find evidence that it is an issue i the library, please provide more info so we can reproduce and fix.

**Comment by MathijsWouters** [comment]
@beto-rodriguez 
![image](https://github.com/user-attachments/assets/318c96c3-3762-440d-a95f-bba15f5b44d6)
atm only 1 label is generated on the x axis called test 1, it should be under it's corresponding bar which is on the left but it always starts in the middle of the page for some reason. If i adjust the unitwidth factor all the labels do show on screen but the first label still appears in the middle not the left side of the page which would be point 0.0 on the axis. 

if that issue can't be fixed i'm wondering how to get rid of the whole x axis and the extra space it takes since right now i made it so there is no x axis or atleast no labels on it but it still populates like a percentage of height on the screen which i would like to use: 

`using LiveChartsCore;
using LiveChartsCore.Measure;
using LiveChartsCore.SkiaSharpView;
using LiveChartsCore.SkiaSharpView.Painting;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;

namespace BeursfuifChiroZevendonk.ViewModels
{
    public partial class BeursPageViewModel : BaseViewModel
    {
        public ObservableCollection<Drink> Drinks => _drinksService.Drinks;
        public ObservableCollection<ISeries> Series { get; set; } = new ObservableCollection<ISeries>();

        private readonly DrinksDataService _drinksService;
        public IEnumerable<Axis> YAxes { get; set; }
        public IEnumerable<Axis> XAxes { get; set; }

        public BeursPageViewModel(DrinksDataService drinksService)
        {
            base.Title = "Beurs";
            _drinksService = drinksService;
            MessagingCenter.Subscribe<App>((App)Application.Current, "PricesUpdated", (sender) => {
                UpdateChart();
            });
            UpdateChart();
        }

        private void InitializeAxes()
        {
            decimal highestMaxPrice = Drinks.Max(drink => drink.MaxPrice) + 0.25m;
            decimal lowestMinPrice = Drinks.Min(drink => drink.MinPrice) - 0.25m;

            YAxes = new Axis[]
            {
        new Axis
        {
            MinLimit = (double)lowestMinPrice,
            MaxLimit = (double)highestMaxPrice,
            ShowSeparatorLines = false,
            LabelsPaint = new SolidColorPaint(SKColors.White),
            TextSize = 20
        }
            };
            XAxes = new Axis[]
{
        new Axis
        {
            IsVisible = false,
        }
};

        }


        public void UpdateChart()
        {
            InitializeAxes();
            var series = new List<ISeries>();

            foreach (var drink in Drinks)
            {
                var columnSeries = new ColumnSeries<decimal>
                {
                    Values = new decimal[] { drink.CurrentPrice },
                    Name = drink.Name,
                    Fill = new SolidColorPaint(SKColor.Parse(drink.DrinkColorHex)),
                    MaxBarWidth = double.NaN, 
                    Padding = 30
                };

                series.Add(columnSeries);
            }

            Series.Clear();
            foreach (var serie in series)
            {
                Series.Add(serie);
            }

            RefreshChart();
        }

        private void RefreshChart()
        {
            OnPropertyChanged(nameof(Series));
            OnPropertyChanged(nameof(YAxes));
            OnPropertyChanged(nameof(XAxes));
        }
    }
}
` with this you code you get 
![image](https://github.com/user-attachments/assets/91de8196-84b0-43ad-8487-3dde853fcb0f)

As seen in the picture the graph takes some white space at the bottom where the x axis and it's labels should be but i don't want it to do that so i can use the space for other things.
